types/index.d.ts

```diff
@@ -1979,8 +1976,10 @@ export declare namespace Knex {
     index(indexName?: string, indexType?: string): ColumnBuilder;
   }
 
-  interface ReferencingColumnBuilder extends ColumnBuilder {
-    inTable(tableName: string): ColumnBuilder;
+  interface ReferencingColumnBuilder extends ColumnBuilder{
+    inTable(tableName: string): ReferencingColumnBuilder;
+    deferrable(type: deferrableType): ReferencingColumnBuilder;
+    withKeyName(keyName: string): ReferencingColumnBuilder;
```

Comment:
extended this ```interface``` and also changed the type of ```inTable``` to ```ReferencingColumnBuilder``` so that all these function can be chained in any order

Reply:
@kibertoad should i also add onUpdate and onDelete to this ```interface``` ? since right now ```onUpdate``` and ```onDelete``` always need to be at the end of the ```foreign/references``` chain

Reply:
if it wouldn't be too hard!